src/commands/art.ts

```diff
@@ -80,20 +61,19 @@ export class ArtCommand extends Command {
 	}
 
 	protected override async execute(interaction: ChatInputCommandInteraction): Promise<number> {
-		const type = interaction.options.getSubcommand(true) as CardLookupType;
-		const input = interaction.options.getString("input", true);
-		const resultLanguage = await this.locales.get(interaction);
-		const inputLanguage = (interaction.options.getString("input-language") as Locale) ?? resultLanguage;
-		// Send out both requests simultaneously
-		const [, card] = await Promise.all([interaction.deferReply(), getCard(this.got, type, input, inputLanguage)]);
-		let end: number;
+		const { type, input, resultLanguage, inputLanguage } = await getCardSearchOptions(interaction, this.locales);
+		const card = await getCard(this.got, type, input, inputLanguage);
 		if (!card) {
-			end = Date.now();
 			useLocale(resultLanguage);
-			await interaction.editReply({ content: t`Could not find a card matching \`${input}\`!` });
+			const reply = await interaction.reply({
+				content: t`Could not find a card matching \`${input}\`!`,
+				fetchReply: true
+			});
+			return replyLatency(reply, interaction);
 		} else {
+			await interaction.deferReply();
```

Comment:
Took me a second to find this, expecting it near the top, but it makes so much more sense to only defer if we know we have a card to fetch. 👍